### PR TITLE
Added babel-cli to distribute ES5 version on npm

### PR DESCRIPTION
**Pull request by holgerd77**
Adresses #111 and #92

### CODE REVIEW

**Comment by nickysemenza** [comment]
I'd love to see this merged 😄

**Comment by holgerd77** [comment]
@nickysemenza This is now in review, hope to have this merged soon.

File: test/defineFields.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 var assert = require('assert')
-var ethUtil = require('../index.js')
+var ethUtil = require('../dist/index.js')
```
Comment (Silur) [inline test/defineFields.js]: maybe tests can be part of the babel space too for some may write these in ES6?

Comment (holgerd77) [reply test/defineFields.js]: Ah, yes, we can think about that, but let's leave this for now.

File: .babelrc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "presets": [
+    [
+      "env"
```
Comment (fanatid) [inline .babelrc]: If publisher will build with latest node this will produce not ES5 output, is not it?

Comment (holgerd77) [reply .babelrc]: Could you be a bit more specific, who is the "publisher", the one who publishes this on npm? And what "latest" node do you mean, which version? What changed in that version, do you have some link on that?

Comment (fanatid) [reply .babelrc]: Sorry, one missed word destroyed my message, I wanted write: `If publisher will build with latest node version...`

I checked https://www.npmjs.com/package/babel-preset-env
> Without any configuration options, babel-preset-env behaves exactly the same as babel-preset-latest (or babel-preset-es2015, babel-preset-es2016, and babel-preset-es2017 together).

Did not knew about this. Sorry for false alarm.

Comment (holgerd77) [reply .babelrc]: No problem, I am not totally the transpilation/babel expert, so better to double check than one check to few. Sorry also for being a bit unconsistent in what Node version we actually support. I think we still have to fine-tune this a bit. I also didn't know how many people still need some ``ES5`` compatible distribution and for what reasons.

So I think it probably makes sense for the various libraries to -yes- drop Node 4 support for the source version to be able to implement with newer and more elegant techniques, but also provide a transpiled distribution version with lower requirements.

Probably also has to be decided/evolved on a case-by-case (library-by-library) basis, depending on the needs of the users, the scope of the library and the compromises which have to be done in respect to security, speed and build/library size.

Actually this topic gets more and more complex for me the more I read on it. Will try to keep pace! 😄

**Comment by gre** [comment]
Is there any chance this will get backport to the `4.x` release? many library still depend on the 4.x , like `ethereumjs-abi` and it will take time for everyone to update to it.

**Comment by holgerd77** [comment]
@gre Hmm, not very probable, sorry, there is so much to do on the various libs, don't think anyone will find the time. Is ``ethereumjs-abi`` your main concern? I would rather look into updating this instead then.

**Comment by gre** [comment]
ok thanks. I've sent a PR for -abi https://github.com/ethereumjs/ethereumjs-abi/pull/59 , i'm going to see which other dependency I have still uses the 4.x.
cheers

**Pull request merged**